### PR TITLE
configure-gpg: Change the way gpg keys is created

### PR DESCRIPTION
**Pull request by mamedin**
The gpg key is created with python's gpg module instead of using gpg
command directly. This is the same way that key is created by the
Archivematica Storage Service.

The `binary` key was removed from `archivematica_src_configure_gpg` dict, now
it is defined in a tasks in role (Bionic=gpg1, Others=gpg)

Connects to https://github.com/archivematica/Issues/issues/1078

### CODE REVIEW

File: tasks/configure-gpg.yml [hunk 1]
```diff
@@ -42,44 +41,30 @@
...
+          input_data = gpg.gen_key_input(
+            name_email = '{{ archivematica_src_configure_gpg.key_mail }}',
+            name_comment = '{{ archivematica_src_configure_gpg.key_comment }}',
+            name_real = '{{ archivematica_src_configure_gpg.key_user }}',
+            passphrase = '{{ archivematica_src_configure_gpg.key_passphrase }}',
+            key_length = '4096',
+            key_type = 'RSA'
+          )
+          key = gpg.gen_key(input_data)
+          print key
```
Comment (ross-spencer) [inline tasks/configure-gpg.yml]: Does this work `print(key)`? It might need to change with Python 3 otherwise.

**Review by ross-spencer — APPROVED** [review]
One comment and from Slack noted this works across distributions, so LGTM! :+1:

File: tasks/configure-gpg.yml (hunk 1 above)
Comment (mamedin) [reply tasks/configure-gpg.yml]: Yes, it prints the fingerprint and we can take it later with: {{ gpg_fingerprint.stdout }}

This is the -vvvvv task output:

``` 
changed: [am111centos] => {
    "changed": true,
    "cmd": [
        "/usr/share/archivematica/virtualenvs/archivematica-storage-service/bin/python"
    ],
    "delta": "0:00:01.033960",
    "end": "2020-01-29 12:37:13.842298",
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/share/archivematica/virtualenvs/archivematica-storage-service/bin/python",
            "_uses_shell": false,
            "argv": null,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": "import os\nimport gnupg\n\ngpg = gnupg.GPG(gnupghome='/var/archivematica/storage_service')\ninput_data = gpg.gen_key_input(\n  name_email = 'monitor+am111centos@artefactual.com',\n  name_comment = 'Ansible Generated',\n  name_real = 'test',\n  passphrase = '',\n  key_length = '4096',\n  key_type = 'RSA')\nkey = gpg.gen_key(input_data)\nprint key\n",
            "stdin_add_newline": true,
            "strip_empty_ends": true,
            "warn": true
        }
    },
    "rc": 0,
    "start": "2020-01-29 12:37:12.808338",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "2C3CA318A954AC32A0E589BECB31C5004C4394ED",
    "stdout_lines": [
        "2C3CA318A954AC32A0E589BECB31C5004C4394ED"
    ]
}
```

Comment (mamedin) [reply tasks/configure-gpg.yml]: @ross-spencer , I just tested with "print(key)" and it works fine. So I'm going to change it.

**Comment by scollazo** [comment]
What's wrong with the keys created from the cli?

**Review by scollazo — COMMENTED** [review]
I'm ok with this, but I'm interested on understanding the root cause of the issue, why keys created from the cli don't work on some envs, but work on others?

**Review by scollazo — APPROVED** [review]

**Comment by mamedin** [comment]
There's no so much info in logs. It is a difficult and time consuming investigation.

**Pull request merged**